h/services/annotation_metadata.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+from sqlalchemy.dialects.postgresql import insert
+from sqlalchemy.orm import Session
+
+from h.models import Annotation, AnnotationMetadata, User
+from h.security import decrypt_jwe_dict
+
+
+class AnnotationMetadataService:
+    def __init__(self, db: Session):
+        self._db = db
+        self._secret = os.environ.get("JWE_SECRET_LMS")
+
+    def set_annotation_metadata_from_jwe(self, annotation: Annotation, jwe: str):
+        """
+        Set the decrypted `jwe` dict as the metadata for `annotation`.
+
+        It will create a new AnnotationMetadata row or update the existing one.
+        """
+        if not annotation.pk and not annotation.user_id:
```

Comment:
This can be scrapped once we complete fill up.

Added a task to the board.

I'l like to to tackle this and not leave it there forgotten. `pk` and `user_id` will be more valuable once every row is guaranteed to have it.